front/lib/resources/data_source_view_resource.ts

```diff
@@ -174,6 +202,12 @@ export class DataSourceViewResource extends ResourceWithVault<DataSourceViewMode
     });
   }
 
+  // Getters.
+
+  get dataSource(): DataSourceResource | undefined {
+    return this.ds;
+  }
+
```

Comment:
is it really possible to have undefined here ? it should always be loaded and exist

Reply:
Yes but the resource flow is not meant for this right now, it requires a bigger refactoring where we can run this in the constructor/entry point.